src/odemis/gui/comp/overlay/world.py

```diff
@@ -347,6 +351,67 @@ def set_icon(feature_icon):
 
             ctx.paint()
 
+class SampleBackgroundOverlay(WorldOverlay):
+    """
+    Overlay to display the contour of samples (aka TEM grid)
+    Displays a list of circles and, at low zoom levels, also show their name in the middle .
+    """
+
+    def __init__(self, cnvs, samples: Dict[str, Tuple[float, float]], sample_radius: float):
+        """
+        cnvs (Canvas): canvas for the overlay
+        sample: Dict of sample name -> center position (X/Y in m)
+        sample_radius: the radius of a sample in m (all samples are shown with the same radius)
+        """
+        super().__init__(cnvs)
+        self._colour = (0.0, 0.42, 0.8, 1.0)  # Blue
+        self._samples = samples
+        self._radius = sample_radius
+        self._labels = {}
+        for name in self._samples:
+            self._labels[name] = self.add_label(name,
+                                                align = wx.ALIGN_CENTRE_HORIZONTAL | wx.ALIGN_CENTER_VERTICAL,
+                                                colour=self._colour[:3],  # no transparency, to support .opacity
+                                                background=(0, 0, 0, 0.5)  # Black semi-transparent
+                                                )
+
+    def draw(self, ctx, shift=(0, 0), scale=1.0):
+        """ Draw the background image by displaying all circles. """
```

Comment:
Would it be good to add parameter explanations in the doc string or do you think it is already self-explanatory?